src/server/command_registry.cc

```diff
@@ -117,7 +117,7 @@ void CommandRegistry::Init(unsigned int thread_count) {
 }
 
 CommandRegistry& CommandRegistry::operator<<(CommandId cmd) {
-  auto k = cmd.name();
+  string k = string(cmd.name());
```

Comment:
The bug was really in line 129, where the result of the trinary expression was of type `string` (because `StrCat()` returns `string`), which would then make `k` point to an invalid `string` in all cases, even when `is_sub_command` is `false`

Reply:
@kostasrim did you chek the restricted command with this changes? If dragonfly starts with ACL as restricted command and someone issues ACL subcommand will it still be restricted?

Reply:
Nice catch!
(unrelated to this change though, but nice!)

Reply:
nice catch, Shahar.

Reply:
@adiholden nope but it should be a straightforward fix. I will take care of it